### PR TITLE
Add basic test cases for the new `ThreadProvider`

### PR DESCRIPTION
**Pull request by git-babel**
### Changelog
This PR will add the test cases for the new `ThreadProvider` and its related hooks.
#### Before
스크린샷 2024-12-09 오전 11 29 53

#### After
스크린샷 2024-12-09 오전 11 28 39

### CODE REVIEW

**Comment by AhyoungRyu** [comment]
Some new tests for these files has been added in here https://github.com/sendbird/sendbird-uikit-react/pull/1274/commits/1fc0c15de56367e5cb254ac8c90575fb49f22f10
Screenshot 2024-12-05 at 5 09 37 PM

File: src/modules/Thread/context/hooks/useResendMessageCallback.ts [hunk 1]
```diff
@@ -35,107 +35,110 @@ export default function useResendMessageCallback({
   pubSub,
 }: StaticProps): (failedMessage: SendableMessageType) => void {
   return useCallback((failedMessage: SendableMessageType) => {
-    if ((failedMessage as SendableMessageType)?.isResendable) {
```
Comment (AhyoungRyu) [inline src/modules/Thread/context/hooks/useResendMessageCallback.ts:38]: Nothing has been updated in these massive change but just added early return block in line number 38.

File: src/modules/Thread/context/hooks/useSendFileMessage.ts [hunk 2]
```diff
@@ -50,37 +50,42 @@ export default function useSendFileMessageCallback({
       const params = onBeforeSendFileMessage?.(file, quoteMessage) ?? createParamsDefault();
       logger.info('Thread | useSendFileMessageCallback: Sending file message start.', params);
 
-      currentChannel?.sendFileMessage(params)
-        .onPending((pendingMessage) => {
+      if (currentChannel == null) {
+        logger.warning('Thread | useSendFileMessageCallback: currentChannel is null. Skipping file message send.');
+        resolve(null);
```
Comment (AhyoungRyu) [inline src/modules/Thread/context/hooks/useSendFileMessage.ts:55]: Same for this file change. Just wrapped with if - else for the better unit testing.

File: src/modules/Thread/context/hooks/useResendMessageCallback.ts (hunk 1 above)
Comment (git-babel) [reply src/modules/Thread/context/hooks/useResendMessageCallback.ts:38]: The changed version looks better. LGTM

File: src/modules/Thread/context/hooks/useSendFileMessage.ts (hunk 2 above)
Comment (git-babel) [reply src/modules/Thread/context/hooks/useSendFileMessage.ts:55]: Same with the above

**Comment by git-babel** [comment]
스크린샷 2024-12-09 오전 11 22 06
test coverage updated.

**Review by AhyoungRyu — APPROVED** [review]

**Pull request merged**